src/main/kotlin/io/github/mojira/arisa/infrastructure/newjira/MapFromJira.kt

```diff
@@ -0,0 +1,124 @@
+package io.github.mojira.arisa.infrastructure.newjira
+
+import com.uchuhimo.konf.Config
+import io.github.mojira.arisa.infrastructure.config.Arisa
+import io.github.mojira.arisa.newdomain.Attachment
+import io.github.mojira.arisa.newdomain.ChangeLogItem
+import io.github.mojira.arisa.newdomain.Comment
+import io.github.mojira.arisa.newdomain.Issue
+import io.github.mojira.arisa.newdomain.Link
+import io.github.mojira.arisa.newdomain.LinkedIssue
+import io.github.mojira.arisa.newdomain.Project
+import io.github.mojira.arisa.newdomain.User
+import io.github.mojira.arisa.newdomain.Version
+import net.sf.json.JSONObject
+import java.text.SimpleDateFormat
+
+private val versionDateFormat = SimpleDateFormat("yyyy-MM-dd")
+
+@Suppress("TooManyFunctions")
+class MapFromJira(
+    val config: Config,
+    val userService: UserService,
+    val attachmentService: AttachmentService
+) {
+
+    fun JiraIssue.toDomain() = Issue(
+        this.key,
+        this.summary ?: "",
+        this.status?.name,
+        this.description ?: "",
+        this.getFieldAsString("environment") ?: "",
+        this.security?.id,
+        this.reporter?.toDomain(),
+        this.resolution?.name,
+        this.createdDate.toInstant(),
+        this.updatedDate?.toInstant(),
+        this.resolutionDate?.toInstant(),
+        this.getFieldAsString(config[Arisa.CustomFields.chkField]),
+        this.getCustomField(config[Arisa.CustomFields.confirmationField]),
+        this.getFieldAsDouble(config[Arisa.CustomFields.linked]),
+        this.getCustomField(config[Arisa.CustomFields.mojangPriorityField]),
+        this.getFieldAsString(config[Arisa.CustomFields.triagedTimeField]),
+        this.project.toDomain(),
```

Comment:
Not sure if that would work, if I recall correctly it would not have the versions list. We would probably have to use `ProjectCache`.